m03_acos/create_acos_exercises.rst

```diff
@@ -0,0 +1,454 @@
+
+.. figure:: /images/gifs/AcosServerLocal.gif
+  :width: 100 %
+  :align: center
+  
+.. warning::
+  * Once you have create the` npm link <https://docs.npmjs.com/cli/link.html>`_ to the folder by typing following the steps 1-2 mentioned above, all the changes made in you exercises will be reflected in the ``node_modules`` folder of the Acos-server
+  * Normally, you will have to stop/start the node module every time you make changes in your exercise. (:kbd:`Ctrl` + :kbd:`c` to stop the module and ``npm start`` in the terminal to start the module).
+
+.. _docker-test:
+Test Acos activities with Docker
+----------------------------------
+After you have compiled your A+ course in your computer for the first time, the Acos docker image is downloaded and installed locally. It means that you can make use of that docker image to test your Acos exercises in a docker container. 
+
+Once you have :ref:`created your activity <create-acos-exercise>`. You should create a ``docker-compose.yml`` file, in the root of your project exercise, with the following information in it. You can also find that file in the zip file provided in this chapter.
+
+.. code-block:: yml
+  :caption: docker-compose.yml file
+  :name: target-name
+  :linenos:
+
+  version: '3'
+
+  services:
+  acos:
+      image: apluslms/run-acos-server
+      ports:
+      - "3000:3000"
+      volumes:
+      - .:/usr/src/acos-server/node_modules/acos-draganddrop-data-management
+
+After adding the ``docker-compose`` file, the structure of your folder should looks like follows.
+
+.. code-block:: bash
+  :caption: The structure of the project should looks like this.
+  :emphasize-lines: 2
+
+  .
+  ├── docker-compose.yml
+  ├── exercises
+  │   ├── data_structure.json
+  │   └── data_structure.xml
+  ├── index.coffee
+  ├── index.js
+  ├── LICENSE
+  ├── node_modules
+  ├── package-lock.json
+  ├── README.md
+  └── static
+      ├── data_folder.png
+      ├── date_folder.png
+      ├── exp_folder.png
+      ├── file.png
+      ├── link_folder2.png
+      ├── link_folder.png
+      ├── project_folder.png
+      ├── readme.png
+      ├── scripts.png
+      └── src_folder.png
+
+
+After creating the ``docker-compose.yml`` file, you should run the following command. This command will allow you to run the Acos server inside the docker image, and see the content in your web browser. You can check that the service is running by opening a web browser and going to the following url http://acos:3000. 
+
+.. code-block:: terminal
+  :caption: Start the Acos server docker container
+
+  docker-compose up
+
+.. image:: /images/gifs/AcosServerDocker.gif
+  :width: 100 %
+  :align: center
+
+.. warning:: 
+  * Every time you make a change in the activity you should stop the docker container by typing :kbd:`Ctrl` + :kbd:`c`. Make the changes to the activity and start the docker container again.
```

Comment:
See notes on warnings above. Although this seems like it actually might be a warning. If so, it would be good to mention what happens one doesn't stop the container by typing Ctrl+C but rather stops it in a different way.